openquake/risklib/scientific.py

```diff
@@ -131,6 +131,10 @@ def __init__(self, vf_id, imt, imls, mean_loss_ratios, covs=None,
                     pass
                 elif lr > 1:
                     raise ValueError('The meanLRs must be <= 1, got %s' % lr)
+                elif cov == 0:
+                    raise ValueError(
+                        'Found a zero coefficient of variation in %s' %
+                        self.covs)
                 elif cov ** 2 > 1 / lr - 1:
                     # see https://github.com/gem/oq-engine/issues/4841
                     raise ValueError(
```

Comment:
We could improve the message to be more informative to something like: "The coefficient of variation %s > %s does not satisfy the requirement `0 < σ < sqrt[μ × (1 - μ)]` .."

Reply:
Meant to add the above comment to Lines 141–142 in the patched file.

Reply:
Message updated